Bottom: 1d1b8a083c800013e937da6d0b24ec6138353705
Top:    36e86064b1de5d4dcfe5b7395e0ec7814a4066f7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-22 02:09:47 +0000

Debug wonky copper polygon generation


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index cd4dfa4..56dddb0 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1436,19 +1436,19 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
       {
         fprintf (stderr, "Accumulating elements from layer %i\n", GetLayerNumber (PCB->Data, layer));
 
-        r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
-        r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
-        r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
+//        r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
+//        r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
+//        r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
         r_search (layer->polygon_tree, &bounds, NULL, polygon_copper_callback, &info);
       }
     END_LOOP;
 
-    fprintf (stderr, "Accumulating pin + via pads\n");
-    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
-    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
+//    fprintf (stderr, "Accumulating pin + via pads\n");
+//    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
+//    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
 #endif
 
-#if 1
+#if 0
     if (group == top_group ||
         group == bottom_group)
       {
@@ -1502,6 +1502,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
                               NULL));
 
 
+    break;
   }
 
 
@@ -1512,7 +1513,11 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
 //  PCB->Data->outline_valid = true;
 //  gui->invalidate_all ();
 
-  poly_Free (&info.poly);
+  /* ASSUME THERE IS A POLYGON WHERE WE KNOW WE PUT ONE... */
+  ((PolygonType *)PCB->Data->Layer[1].Polygon->data)->Clipped = info.poly;
+  gui->invalidate_all ();
+
+//  poly_Free (&info.poly);
 
   return objects;
 }
diff --git a/src/polygon1.c b/src/polygon1.c
index 5c09126..2372153 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -39,6 +39,7 @@
       are marked
 */
 
+#undef NDEBUG
 #include	<assert.h>
 #include	<stdlib.h>
 #include	<stdio.h>
@@ -109,13 +110,13 @@ int vect_inters2 (Vector A, Vector B, Vector C, Vector D, Vector S1,
 
 #define error(code)  longjmp(*(e), code)
 
-#undef DEBUG_INTERSECT
-#undef DEBUG_LABEL
-#undef DEBUG_ALL_LABELS
-#undef DEBUG_JUMP
-#undef DEBUG_GATHER
-#undef DEBUG_ANGLE
-#undef DEBUG
+#define DEBUG_INTERSECT
+#define DEBUG_LABEL
+#define DEBUG_ALL_LABELS
+#define DEBUG_JUMP
+#define DEBUG_GATHER
+#define DEBUG_ANGLE
+#define DEBUG
 #ifdef DEBUG
 #define DEBUGP(...) pcb_fprintf(stderr, ## __VA_ARGS__)
 #else
